### ISSUE TITLE
When a deprecated direct update command is used, inform the user of the recommended update method

### ISSUE THREAD

**Issue by andrewdavidwong** [QubesOS/qubes-issues#8076]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### The problem you're addressing (if any)

Many Qubes users have become accustomed, over the years, to using direct commands to perform updates, such as `qubes-dom0-update`, `dnf update`, and `apt update`. Even though this practice was deprecated long ago for security reasons (see #6275, https://github.com/QubesOS/qubes-posts/pull/79, and https://github.com/QubesOS/qubes-doc/commit/9eea4db0a62b7b7f70e3199e2abc095ada780cda), some users likely *still* aren't aware of it, since they would have to read the documentation or GitHub to learn of it. There is nothing in Qubes OS software itself that ever hints to them that they're using a deprecated update method.

### The solution you'd like

Here's a simple version of the idea: When a direct update command is used, print a brief notification, e.g.:

```
The use of <DIRECT_UPDATE_COMMAND> and other direct update commands is no longer recommended. Please consider using the Qubes Update tool instead. For more information, please see: https://www.qubes-os.org/doc/how-to-update/
```

(Where `<DIRECT_UPDATE_COMMAND>` is the direct update command the user just executed, such as `qubes-dom0-update`, `dnf update`, or `apt update`.)

If the recommended update method changes again with #7443, or if the name of the preferred tool changes, then this notification should be updated accordingly.

### The value to a user, and who that user might be

Users who aren't aware that they're using a deprecated update method will have a better chance of becoming aware *without* having to read documentation, announcements, or discussions outside of the system itself. In this way, this proposal is actually better than the PSA I had wanted to make (https://github.com/QubesOS/qubes-posts/pull/79).

**Comment by andrewdavidwong** [comment]
@marmarek, are you saying that the recommended update method may effectively go back to being the direct update commands and hence that what I'm proposing here may not be necessary?

**Comment by marmarek** [comment]
In case of dom0 - yes.

**Comment by andrewdavidwong** [comment]
> https://www.qubes-os.org/doc/how-to-update/ used to say to run qubes-dom0-update after to make sure nothing was missed and that the salt, wasn't falsely reporting a successful update. Maybe this issue has been fixed and the 2nd line in my script is no longer needed since it is no longer in the docs. Can anyone confirm?

That's correct. See https://github.com/QubesOS/qubes-doc/commit/5ebedc545d7e2cbea003beb54e25af6c09a04825 and https://github.com/QubesOS/qubes-issues/issues/6585 for details.

> As mentioned qubes-dom0-update is still in use in higher-level tools and sounds like it will remain in use to apply extra fixes even if it is being migrated away from.

That is not my understanding. My understanding is that we will *not* migrate away from `qubes-dom0-update` *precisely because* it will be able to apply the "extra fixes" that previously (and currently?) only Salt can do.
